handler/websocket.go

```diff
@@ -63,33 +63,42 @@ func connectWs(exec graphql.ExecutableSchema, w http.ResponseWriter, r *http.Req
 		cfg:    cfg,
 	}
 
-	if !conn.init() {
+	initPayload, ok := conn.init()
+	if !ok {
 		return
 	}
 
-	conn.run()
+	conn.run(initPayload)
 }
 
-func (c *wsConnection) init() bool {
+func (c *wsConnection) init() (initPayload graphql.InitPayload, ok bool) {
```

Comment:
wsConnection is stateful per connection, the init payload could be added to there to avoid passing it around?

Reply:
That's a good idea, actually. I'm not sure why I didn't think of that.